chia/full_node/mempool_manager.py

```diff
@@ -81,14 +82,38 @@ def validate_clvm_and_signature(
     return None, bytes(result), new_cache_entries
 
 
+def compute_assert_height(
+    removal_coin_records: Dict[bytes32, CoinRecord],
+    conds: SpendBundleConditions,
+) -> uint32:
+    """
+    Computes the most restrictive height assertion in the spend bundle. Relative
+    height assertions are resolved using the confirmed heights from the coin
+    records.
+    """
+
+    height: uint32 = uint32(conds.height_absolute)
+
+    for spend in conds.spends:
+        if spend.height_relative is None:
+            continue
+        h = uint32(removal_coin_records[bytes32(spend.coin_id)].confirmed_block_index + spend.height_relative)
```

Comment:
I'm not positive but should this have a special case for 0?  I seem to recall that `(ASSERT_HEIGHT_RELATIVE 0)` would result in being unable to spend the coin ephemerally basically acting as a `(ASSERT_HEIGHT_RELATIVE 1)`.  Am I mistaken in recalling this?

Reply:
Or is it actually +1 because the timestamp is based off of the previous block maybe?

Reply:
your recollection is correct. the special case is the check on line 98. a relative height of 0 means the coin has to be spent in at least the block after it was confirmed. the `assert_height` being computed here has the same semantics as the `ASSERT_HEIGHT_ABSOLUTE`, meaning the block height has to be *greater than* the specified one for the spend to be valid. i.e. the same block is still too soon.